text/0000-pack-cache-options.md

```diff
@@ -0,0 +1,224 @@
+
+* By default, `pack build` would create two named volumes
+
+<!-- - Why should we do this?
+- What use cases does it support?
+- What is the expected outcome? -->
+
+##### Use Cases
+
+* As a Pack user, I would like to build using a docker daemon while storing _`build` cache_ as an image (aka _`cache-image`_) in a registry. ([buildpacks/pack#1024](https://github.com/buildpacks/pack/issues/1024#issuecomment-762901676))
+* As a Pack user using `pack` on BitBucket Pipelines, I would like to be able to store the _`build` cache_ and _`launch` cache_ on the file system. ([bitbucket discussion](https://community.atlassian.com/t5/Bitbucket-discussions/Using-CNCF-Buildpacks-on-Bitbucket-Pipelines/td-p/1516690))
+* As a Pack user, I would like to be able to specify the name of the volumes used by pack for caching. ([buildpacks/pack#665](https://github.com/buildpacks/pack/issues/665))
+
+# What it is
+[what-it-is]: #what-it-is
+
+<!-- This provides a high level overview of the feature.
+
+- Define any new terminology.
+- Define the target persona: buildpack author, buildpack user, platform operator, platform implementor, and/or project contributor.
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users. -->
+
+A new variadic flag that enables configuring different cache types geared towards app developers or operators that are using pack in very specific (and sometimes restrictive) ways that require them to configure caching in a more advanced fashion.
+
+The options available via this new flag would have sensible defaults based on the current behaiviour with the exception of the `--cache-image` which would be deprecated and later removed to reduce confusion. Overall UX would remain the same while allowing for greater flexibility.
+
+An additional benefit of this format is that it can later be expanded on if more options for a specific format were deamed necessary.
+
+```
+pack build --cache 'type=<type>;format=<format>;<additional-options>' --cache ...
+```
+
+### `cache` flag options
+* _**type**_ - The type of cache to configure [`build`, `launch`] (default: `build`).
+* _**format**_ - The format in which to store that cache [`bind`, `image`, `volume`] (default: `volume`). 
+
+###### Format specific options
+
+* `bind`
+    * _**source**_ - Path to the local directory to use as cache.
+* `image`
+    * _**name**_ - Name of the image to use to save as cache (including tag).
+* `volume`
+    * _**name**_ - Name of the volume to use to save as cache. (default: `pack-<type>-<app-uuid>`)
+
+##### Examples
+
+```bash
+# specify named volume build cache
+pack build --cache 'type=build;format=volume;name=my-build-cache'
+
+# specify bind mount cache (relative to current working directory)
+pack build --cache 'type=build;format=bind;source=./my-build-cache'
```

Comment:
```suggestion
pack build --cache 'type=build;format=bind;name=./my-build-cache'
```
Where possible, I'd like to keep the same option names for all options (if that's sensible), to minimize the loss factor when switching cache types.

Reply:
I can't seem to find @ekcasey's comment that inspired the change from `format=dir;path=./my-build-cache` to match Docker's [`mount`](https://docs.docker.com/storage/bind-mounts/#choose-the--v-or---mount-flag) format.

Personally, I think it's more confusing to use `name` because it doesn't clearly depict what "name" means in this context whereas `source` is pretty clear, especially if you are familiar with `mount`.